File: tests/java/org/elastos/hive/PromotionServiceTest.java
package org.elastos.hive;

import org.elastos.hive.config.TestData;
import org.elastos.hive.service.PromotionService;
import org.junit.jupiter.api.*;

@Disabled
@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
class PromotionServiceTest {

	private static PromotionService promotionService;

	@BeforeAll public static void setup() {
		Assertions.assertDoesNotThrow(()->promotionService = TestData.getInstance().newBackup().getPromotionService());
	}

	@Test void testPromote() {
		Assertions.assertDoesNotThrow(()->promotionService.promote().get());
	}

}

File: tests/src/promotionservice.test.ts
import { TestData } from "./config/testdata";
import {PromotionService, VaultSubscription} from "../../src";

describe.skip("test database services", () => {
    let testData:TestData;
    let promotionService: PromotionService;

    beforeAll(async () => {
        testData = await TestData.getInstance("promotionservice.tests");
        const vaultSubscription = new VaultSubscription(
            testData.getUserAppContext(),
            testData.getProviderAddress());
        try {
            await vaultSubscription.unsubscribe(true);
        } catch (e) {
            // do nothing.
        }
        promotionService = testData.newBackup().getPromotionService();
    });

    test("testPromote", async () => {
        await promotionService.promote();
    });
});
